### DISCUSSION TITLE
About roles problem

### DISCUSSION THREAD

**Discussion by peteresteris** [strapi/strapi#6464] [Legacy Topics]
I have 2 users and the standard role configuration. One of them is administrator and the other one is confirmed and has the authenticated role.
![Clipboard01](https://user-images.githubusercontent.com/53539821/83339245-dd9d7000-a299-11ea-9b58-25867ec6da26.jpg)
The problem is when I authenticate as anyone the role applied is public. I know that because I have public role very restricted; certainly it only can find data. When I try to post or put new data I obtain 403 Forbidden. As soon as I enable create for the public role on that content, the app works. 
It's good to know that administrator has all rights and authenticated has creation an update rights on the content I am trying to modify.
Thanks a lot for be there. This community is a great help.
Here some example:

_createOwner(jwt: string, userId: number) {
    const httpSecOptions = this.getSecHeader(jwt);
    const postData = {
      'cars_count': 0,
      'status': 2,
      'user': userId,
    };
    this.strapiCnx.setTable('/owners');
    this.strapiCnx.setFilter('');
    return this.http.post<Owner>(this.strapiCnx.getUrlStr(), postData, httpSecOptions);
  }

seekOwner() { // throw new Error("Method not implemented.");
    this.strapi.userOwnership(this.jwt, this.userID).subscribe(owner => {
      if (owner !== null && owner !== undefined && owner.length > 0) {
        this.ownership = owner[0];
      } else {
        this.strapi.createOwner(this.jwt, this.userID).subscribe(own => {
          this.ownership = own;
        }, err2 => {
          console.error('Error creating ownership', err2);
        });
      }
    }, err => {
      console.error('Error getting ownership.', err);
    });
  }_
The results:
_headers: HttpHeaders {normalizedNames: Map(0), lazyUpdate: null, lazyInit: ƒ}
message: "Http failure response for http://192.168.0.50:1337/owners: 403 Forbidden"
name: "HttpErrorResponse"
ok: false
status: 403
statusText: "Forbidden"
url: "http://192.168.0.50:1337/owners"_

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by peteresteris** [comment]
It seems the problem is on my Ionic code.
I checked that Bearer token is not added.
This is my header method:
_private httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json; charset=utf-8'
    })
  };

  public getSecHeader(jwt: string) {
    if (this.httpOptions.headers.has('Authorization')) {
      this.httpOptions.headers.set('Authorization', 'Bearer ' + jwt);
    } else {
      this.httpOptions.headers.append('Authorization', 'Bearer ' + jwt);
    }
    return this.httpOptions;
  }_

**Comment by peteresteris** [comment]
Ok, I solved the problem. It was produced by my HttpHeaders implementation. I changed the method this way:
_public getSecHeader(jwt: string) {
    const httpOptions = {
      headers: new HttpHeaders({
        'Accept': 'application/json',
        'Content-Type': 'application/json; charset=utf-8',
        'Authorization': 'Bearer ' + jwt,
      })
    };
    return httpOptions;
  }_
